File: core/src/com/symbol/ecs/component/map/SquareSwitchComponent.java
package com.symbol.ecs.component.map;

import com.badlogic.ashley.core.Component;
import com.badlogic.gdx.utils.Pool;

public class SquareSwitchComponent implements Component, Pool.Poolable {

    public int targetId = 0;
    public boolean toggle = true;

    @Override
    public void reset() {
        targetId = 0;
        toggle = true;
    }

}

File: core/src/com/symbol/ecs/component/map/SquareSwitchComponent.kt
package com.symbol.ecs.component.map

import com.badlogic.ashley.core.Component
import com.badlogic.gdx.utils.Pool

class SquareSwitchComponent : Component, Pool.Poolable {

    var targetId: Int = 0
    var toggle: Boolean = true

    override fun reset() {
        targetId = 0
        toggle = true
    }

}